README.md

```diff
@@ -1,134 +1,121 @@
 
-There are 7 steps to configure the `SmartSelect`:
+There are 7 steps to configure a `SmartSelect`:
 
-1. Import the SmartSelect and SmartSelect.Utilities module and expose the `SearchUnion` type. We will get back to why we need the `SearchUnion` later.
+1. Import the select and add it to your model providing the datatype of the data to be selected.
 
 ```elm
-import SmartSelectSingle as SingleSelect
-import SmartSelect.Utilities exposing (SearchUnion(..))
-```
-
-2. Add the smart select to the model providing the datatype of the data to be selected and initialize it in the model init
+import SingleSelectLocal as Select
 
-```elm
 type alias Product =
     { id : Int
     , name : String
     , price : String
     }
 
 type alias Model =
-    { ...
-    , singleSelect : SingleSelect.SmartSelect Product
+    { products : List Product
+    , select : Select.SmartSelect Msg Product
     }
-
-init : ( Model, Cmd Msg )
-init =
-    ( { ...
-      , singleSelect = SingleSelect.init
-      }
-    )
 ```
 
-3. One message needs to be defined indicating that an update to the smart select has been triggered and needs to be handled. This message expects a tuple containing a `SingleSelect.Msg` and the selection state. **_Note:_** The selection state for the `SmartSelectMulti` is a List of selected entities as opposed to a Maybe entity for `SmartSelectSingle`.
+2. Define a `Msg` to handle updates from the select.
 
 ```elm
 type Msg
     = ...
-    | HandleSelectUpdate ( SingleSelect.Msg Product, Maybe Product )
+    | HandleSelectUpdate (Select.Msg Product)
 ```
 
-4. Configure the settings for the smart select. The `Settings` type as defined in the `SmartSelectSingle` module is below:
+3. Configure the `Settings` for the select. The `internalMsg` field takes a function that expects a `Select.Msg` and returns the `Msg` we defined in step 2. Please refer to the module documentation for more information on the settings.
 
 ```elm
-type alias Settings msg a =
-    { internalMsg : ( Msg a, Maybe a ) -> msg
-    , optionType : String
-    , optionLabel : a -> String
-    , optionDescription : a -> String
-    , searchFn : SearchUnion a
-    , debounceDuration : Float
-    , characterSearchThreshold : Int
-    , closeOnSelect : Bool
+selectSettings : Select.Settings Msg Product
+selectSettings =
+    { internalMsg = \msg -> HandleSelectUpdate msg
+    , searchFn =
+        \searchText products ->
+            List.filter (\product -> String.contains (String.toLower searchText) (String.toLower product.name)) products
+    , optionType = "Product"
+    , optionLabel = \product -> product.name
+    , optionDescription = \product -> product.price
+    , optionsContainerMaxHeight = 300
+    , optionHeight = 50
```

Comment:
Maybe we can split the settings into `Settings` and `CustomSettings` so you only need pass in internalMsg, optionTextFn.

But we can add that later. And get it released now.

Reply:
I agree we should do it later if at all. I don't really see the settings getting more fields at this point and I like that everything an implementer needs to do is in one place.

Reply:
I just feel like there is way to much stuff that you have to think about. And if you decide you want to change the default from 50 to 30 everywhere it is a lot of work. Ideally the smart select should be as easy to use as the native select. But currently it is much more complicated and there is just no reason for it.

Reply:
I'm not sure how introducing another set of settings makes things simpler especially for those who wish to customize those values.

My understanding of what we would need to do to make this happen is the following:
1) define a new `SettingsDefinition` type alias... this would have all the fields a user absolutely needs to define.
2) define a `generateSettings` function that takes in `SettingsDefinition` and produces `Settings` by setting the non-essential settings fields with default values.
3) For users who wish to customize the non-essential fields they then need to expand the `Settings` returned from `generateSettings`.

It is not terrible, but it is two extra steps just so the implementer doesn't need to be bothered with understanding two fields which they are likely to want to change anyway.

Reply:
Even when you have special stuff you want to get something working quickly without needing to understand everything. And I think in most cases people don't want to change those things but who knows.
We don't need to do 1 and the user can change fields or just create the whole thing on her own.

The function would be 
```
generateSettings : { updateMsg: Msg, viewData, a -> String }
```

Reply:
Yeah, lets keep it like this for now and maybe come up with a simpler API after using the package in a few place. Maybe better defaults emerge or maybe not.